scipy/optimize/_tnc.py

```diff
@@ -409,6 +412,10 @@ def _minimize_tnc(fun, x0, args=(), jac=None, bounds=None,
 
     if maxfun is None:
         if maxiter is not None:
+            warnings.warn(
+                        "`maxiter` has been deprecated in favor of `maxfun`. "
+                        "It will be completely removed in SciPy 1.11.0.",
+                        DeprecationWarning)
```

Comment:
```suggestion
            warnings.warn(
                "`maxiter` has been deprecated in favor of `maxfun`"
                " and will be removed in SciPy 1.11.0.",
                DeprecationWarning, stacklevel=3
            )
```

Reply:
Thanks for the suggestions @tupui. I will update the code and solve the failures in CI.